web/ts/stream-search.ts

```diff
@@ -0,0 +1,77 @@
+enum SearchDomain {
+    VideoSections = "sections",
+    Streams = "streams",
+    Chats = "chats",
+}
+
+export class StreamSearch {
+    readonly courseId: number;
+
+    timeout: NodeJS.Timeout;
+    loading: boolean;
+
+    results: object[];
+
+    constructor(courseId: number, slug: string) {
+        this.courseId = courseId;
+        this.results = [];
+        this.loading = false;
+    }
+
+    getResults(results: number[][]): object[] {
+        const all = []; // create key value pair from array of array
```

Comment:
This should be a better documentation of this function
```suggestion 
        // flatten array of array, preserving only the first occurrence of each streamId
        const all = [];
```